include/pybind11/pybind11.h

```diff
@@ -250,6 +250,7 @@ class cpp_function : public function {
         /* Generate a proper function signature */
         std::string signature;
         size_t type_index = 0, arg_index = 0;
+        std::string self_type("object");
```

Comment:
I've included a default value here because, although unusual, it is a valid to use `py::is_operator()` on a function. In which case this variable would never be set by the loop below.
Using `object` as the default value makes sense because an operator method/function will accept any type when it falls back to returning `NotImplemented`: https://github.com/pybind/pybind11/blob/d54d6d8c613914a7063897d5ddddc62348ffe74a/include/pybind11/pybind11.h#L738-L740

Reply:
This comment is no longer relevant.